src/main/java/org/fogbowcloud/arrebol/execution/creator/K8sWorkerCreator.java

```diff
@@ -0,0 +1,78 @@
+package org.fogbowcloud.arrebol.execution.creator;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.UUID;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.arrebol.execution.Worker;
+import org.fogbowcloud.arrebol.execution.k8s.K8sConfiguration;
+import org.fogbowcloud.arrebol.execution.k8s.K8sTaskExecutor;
+import org.fogbowcloud.arrebol.execution.k8s.client.DefaultK8sClient;
+import org.fogbowcloud.arrebol.execution.k8s.client.K8sClient;
+import org.fogbowcloud.arrebol.execution.k8s.resource.DefaultK8sClusterResource;
+import org.fogbowcloud.arrebol.execution.k8s.resource.K8sClusterResource;
+import org.fogbowcloud.arrebol.models.configuration.Configuration;
+import org.fogbowcloud.arrebol.models.specification.Specification;
+import org.fogbowcloud.arrebol.processor.spec.WorkerNode;
+import org.fogbowcloud.arrebol.resource.MatchAnyWorker;
+import org.fogbowcloud.arrebol.utils.AppUtil;
+
+public class K8sWorkerCreator implements WorkerCreator {
+
+	private final K8sConfiguration configuration;
+	private final Logger LOGGER = Logger.getLogger(K8sWorkerCreator.class);
+
+	public K8sWorkerCreator(Configuration configuration) throws Exception {
+		this.configuration = new K8sConfiguration(configuration);
+	}
+
+	@Override
+	public Collection<Worker> createWorkers(Integer poolId) {
+		Collection<Worker> workers = new LinkedList<>();
+		String address = configuration.getAddress();
+		int poolSize = configuration.getCapacity();
+		String namespace = configuration.getNamespace();
+		String volumeName = configuration.getVolumeName();
+		for (int i = 0; i < poolSize; i++) {
+			LOGGER.info("Creating k8s worker with address=" + address);
+			Worker worker = createK8sWorker(poolId, address, namespace, volumeName);
+			workers.add(worker);
+		}
+
+		return workers;
+	}
+
+	@Override
+	public Collection<Worker> createWorkers(Integer poolId, WorkerNode workerNode) {
+		Collection<Worker> workers = new LinkedList<>();
+		String namespace = configuration.getNamespace();
+		String volumeName = configuration.getVolumeName();
+        for(int i = 0; i < workerNode.getPoolSize(); i++){
+            LOGGER.info("Creating k8s worker with address=" + workerNode.getAddress());
+            Worker worker = createK8sWorker(poolId, workerNode.getAddress(), namespace, volumeName);
+            workers.add(worker);
+        }
+        return workers;
+	}
+
+	private Worker createK8sWorker(Integer poolId, String address, String namespace, String volumeName) {
+		String id = "k8s-executor-" + UUID.randomUUID().toString();
+		K8sClusterResource k8sClusterResource = new DefaultK8sClusterResource(id, address);
+		Specification resourceSpec = null;
```

Comment:
just to make a note. Maybe we want it to be filled with the requirements we talked about, right? I mean, I think that this is the place we want to specify them.

Reply:
As we discussed before, there are two ways to specify the amount of CPU and RAM consumed by the POD container in the k8s cluster worker, the minimum (`requests`) and the maximum (`limits`), I believe that here can enter the maximum information, but the minimum does not fit, as the client that details this information. I will add a TODO to add something to that specification in the future.